src/main/java/core/basesyntax/dataservices/DataConverterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.dataservices;
+
+import core.basesyntax.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convert(List<String> transactions) {
+        List<FruitTransaction> transactionList = new ArrayList<>();
+        for (String transaction : transactions) {
+            String[] parts = transaction.split(",");
+            FruitTransaction.Operation operation = operationFromCode(parts[0]);
+            String fruit = parts[1];
+            int quantity = Integer.parseInt(parts[2]);
+            transactionList.add(new FruitTransaction(operation, fruit, quantity));
```

Comment:
It's good practice to validate the input data before processing it. Ensure that the parts array has the expected number of elements before accessing them to avoid potential `ArrayIndexOutOfBoundsException`.